File: src/main/java/com/kevinmatthiesenfinal/kevinmatthiesenfinal/service/ModelService.java
package com.kevinmatthiesenfinal.kevinmatthiesenfinal.service;

import java.util.List;

import com.kevinmatthiesenfinal.kevinmatthiesenfinal.entity.Model;

/**
 * Handles logic concerning vehicle models.
 * 
 * @author kmatthiesen
 *
 */
public interface ModelService {

	/**
	 * Adds a new model to the database.
	 * 
	 * @param model
	 *            The model to be added to the database.
	 * @return The model that was added to the database.
	 */
	public Model add(Model model);

	/**
	 * Gets a list of all models from the database.
	 * 
	 * @return The list of all models from the database.
	 */
	public List<Model> getModels();

}

File: src/main/webapp/public/app/service/modelService.js
angular.module("final-project").service("modelService", ["$http", function($http){

    var modelEndPoint = "/model";

    // Gets a list of all current vehicle models.
    function getModels() {

        return $http.get(modelEndPoint);

    }

    // Creates a new vehicle model.
    function newModel(model) {

        return $http.post(modelEndPoint, model);

    }

    // List of available functions.
    return {

        getModels: getModels,
        newModel: newModel

    };

}]);
